experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -680,3 +680,78 @@ function Base.iterate(iter::WeylOrbitIterator, state::WeylIteratorNoCopyState)
   (wt, _), state = it
   return deepcopy(wt), state
 end
+
+################################################################################
+# converting to permutations
+
+function perm(w::WeylGroupElem)
+  coxeter_type, n = root_system_type(root_system(parent(w)))[1]
+	@req coxeter_type == :A "Weyl group is not the symmetric group"
+	G = symmetric_group(n + 1)
+	return reduce(*, [cperm(G, [i, i + 1]) for i in word(w)]; init = cperm(G))
+end
+
+function (W::WeylGroup)(p::PermGroupElem)
+  coxeter_type, n = root_system_type(root_system(W))[1]
+	@req coxeter_type == :A && degree(parent(p)) == n+1 "Weyl group is not Sym($(n+1))."
+  word = UInt8[]
+  for cycle in cycles(p)
+    transpositions = [
+      sort([c, cycle[i + 1]]) for (i, c) in enumerate(cycle) if i < length(cycle)]
+    for t in transpositions
+      word = reduce(vcat, [
+        [i for i in t[1]:t[2] - 1],
+        [i for i in reverse(t[1]:t[2] - 2)],
+        word])
+    end
+  end
+  return W(word)
+end
```

Comment:
Please re-write these two functions into proper isomorphisms. I already did something similar for `FPGroup`, see https://github.com/oscar-system/Oscar.jl/blob/0b52e65738d6ce868f59f14259b9a471c92abc78/experimental/LieAlgebras/src/WeylGroup.jl#L449